### PR TITLE
Output dependency metrics in json format

### PR DESCRIPTION
**Pull request by nasifimtiazohi**

### CODE REVIEW

File: depdive/src/lib.rs [hunk 1]
```diff
@@ -97,9 +103,57 @@ pub struct DepDiveReport {
...
+
 pub struct DependencyAnalyzer;
 
 impl DependencyAnalyzer {
+    pub fn get_dep_pacakge_metrics_from_path(path: &Path, only_direct: bool) -> Result<String> {
+        let graph = MetadataCommand::new().current_dir(path).build_graph()?;
+        Self::get_dep_pacakge_metrics(&graph, only_direct)
+    }
+
+    fn get_dep_pacakge_metrics(graph: &PackageGraph, only_direct: bool) -> Result<String> {
```
Comment (bmwill) [inline depdive/src/lib.rs:49]: might want to name this fn in such a way that indicates it outputs Json

File: depdive/src/lib.rs [hunk 2]
```diff
@@ -235,6 +235,18 @@ impl DependencyAnalyzer {
 pub struct DependencyGraphAnalyzer;
 
 impl DependencyGraphAnalyzer {
+    pub fn get_code_metrics_from_path(path: &Path, only_direct: bool) -> Result<String> {
+        let graph = MetadataCommand::new().current_dir(path).build_graph()?;
+        Self::get_code_metrics(&graph, only_direct)
+    }
+
+    fn get_code_metrics(graph: &PackageGraph, only_direct: bool) -> Result<String> {
```
Comment (bmwill) [inline depdive/src/lib.rs:92]: Same here

File: depdive/Cargo.toml [hunk 3]
```diff
@@ -8,7 +8,6 @@ edition = "2018"
 
 [dependencies]
 anyhow = "1.0.38" # error handling library
-tabled = "0.2.1"
```
Comment (bmwill) [inline depdive/Cargo.toml:11]: Whats the rationale for wanting to remove this?

Comment (nasifimtiazohi) [reply depdive/Cargo.toml:11]: rationale is - There are too many fields for a pretty `tabled` output which is no better than just `json` output. In midpoint demo, I just chose a subset of the fields to that doesn't break in the monitor screen but that ain't a feature. I think we can make it a future work on how to present a higher-level abstraction of these data based on idea here. #103

**Review by bmwill — DISMISSED** [review]

**Review by bmwill — APPROVED** [review]

**Pull request closed without merging**